frontend/src/metabase/containers/DataPicker/CardPicker/CardPickerView.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useCallback, useMemo } from "react";
+import _ from "underscore";
+
+import SelectList from "metabase/components/SelectList";
+
+import { canonicalCollectionId } from "metabase/collections/utils";
+
+import type { ITreeNodeItem } from "metabase/components/tree/types";
+import type { Collection } from "metabase-types/api";
+import type Table from "metabase-lib/lib/metadata/Table";
+
+import type { DataPickerSelectedItem, VirtualTable } from "../types";
+import PanePicker from "../PanePicker";
+
+import { StyledSelectList } from "./CardPicker.styled";
+
+interface CardPickerViewProps {
+  collectionTree: ITreeNodeItem[];
+  virtualTables?: VirtualTable[];
+  selectedItems: DataPickerSelectedItem[];
+  targetModel: "model" | "question";
+  onSelectCollection: (id: Collection["id"]) => void;
+  onSelectedVirtualTable: (id: Table["id"]) => void;
+  onBack?: () => void;
+}
+
+function TableSelectListItem({
+  table,
+  targetModel,
+  isSelected,
+  onSelect,
+}: {
+  table: VirtualTable;
+  targetModel: "model" | "question";
+  isSelected: boolean;
+  onSelect: (id: Table["id"]) => void;
+}) {
+  const icon = useMemo(() => {
```

Comment:
There is no benefit in using `useMemo` here. WDYT about converting this to a function and calling it from here instead?